### ISSUE TITLE
VCS Legend issues

### ISSUE THREAD

**Issue by chaosphere2112** [UV-CDAT/vcs#22]
Looks like there are a couple of issues with the legend (at least when using boxfill).

[This notebook illustrates most of the issues I found](https://github.com/chaosphere2112/uvcdat-site/blob/93a5649b0d64e8f324fc1289b218b6fa0e2a92a8/iPython/boxfill/boxfill.ipynb), but I'll break it down individually as well.

### boxfill.legend label placement is wrong

When you assign values to it, it should use that dictionary to draw tickmarks on the legend at the appropriate point for the values used as the keys; this should work fine:
```
import vcs
box_data = [range(0, 100, 20) for _ in range(5)]
boxfill = vcs.createboxfill()
boxfill.boxfill_type = "custom"
boxfill.levels = [0, 10, 30]
boxfill.legend = {1: "one", 16: "sixteen", 25: "twentyfive"}
x = vcs.init()
x.plot(box_data, boxfill)
```

![legend_placement](https://cloud.githubusercontent.com/assets/718512/19327113/421e01da-9082-11e6-8f11-d567c6709ff5.png)

### Legend will draw as many sections as it can

When you have more fillarea* items than levels, weird stuff starts to happen with the legend (though the data looks fine).

```
import vcs, cdms2
x = vcs.init()
f = cdms2.open(vcs.sample_data + '/clt.nc')
s = f('clt')
box = vcs.createboxfill()
box.boxfill_type = "custom"
box.fillareacolors = [1, 2, 3, 4]
box.fillareaindices = [2, 3, 4, 5]
box.fillareastyle = "hatch"
box.levels = [0, 50, 100]
x.plot(s, box)
```

![wat](https://cloud.githubusercontent.com/assets/718512/19326981/b7d8bb28-9081-11e6-84b0-b5eff6596581.png)

At some point I was also able to make legend sections draw over themselves, but I haven't been able to reproduce that one yet.

**Comment by aashish24** [comment]
@chaosphere2112 okay if @danlipsa look into this?

**Comment by chaosphere2112** [comment]
@aashish24 Yup, that's fine with me. I don't really work on VCS proper anymore, besides miscellaneous bug fixes I need in a hurry.

**Comment by doutriaux1** [comment]
First one is fixed second one is still true, even weirder when plotting the data is first REALLY weird but x.png fixes it

**Comment by scottwittenburg** [comment]
This can be fixed where we draw the legend.
